`ColumnAlloc` resuse bytes and unsafe.`BytesToString` caused data error.

``ColumnAlloc.New`` reuse byte slice and use unsafe.`BytesToString` caused data error.

More deailt in these test case:

```
func TestColumnReuse(t *testing.T) {
	db := pg.Connect(&pg.Options{
		User: "postgres",
	})

	err := createSchema(db)
	panicIf(err)

	user1 := &User{
		Name:   "admin",
		Emails: []string{"admin1@admin", "admin2@admin"},
	}
	err = CreateUser(db, user1)
	panicIf(err)

	err = CreateUser(db, &User{
		Name:   "root",
		Emails: []string{"root1@root", "root2@root"},
	})
	panicIf(err)

	story1 := &Story{
		Title:    "Cool story",
		AuthorId: user1.Id,
	}
	err = CreateStory(db, story1)
	panicIf(err)

	users := []map[string]interface{}{}
	stories := []map[string]interface{}{}

	ctx := context.Background()
	_, err = db.QueryContext(ctx, &users, `SELECT * FROM users`)
	panicIf(err)

	_, err = db.QueryContext(ctx, &stories, `SELECT * FROM stories`)
	panicIf(err)

	for _, user := range users {
		ks := []string{}
		for k, _ := range user {
			ks = append(ks, k)
		}
		require.ElementsMatch(t, []string{"name", "id", "email"}, ks)
	}
	for _, story := range stories {
		ks := []string{}
		for k, _ := range story {
			ks = append(ks, k)
		}
		require.ElementsMatch(t, []string{"name", "id", "author_id"}, ks)
	}
}
```

Result, plz notice than column's name is error:
```
➜  pg git:(v10) ✗ TZ= PGSSLMODE=disable go test ./... -run=TestColumnReuse
--- FAIL: TestColumnReuse (0.02s)
    example_db_query_test.go:100:
                Error Trace:    example_db_query_test.go:100
                Error:          elements differ

                                extra elements in list A:
                                ([]interface {}) (len=1) {
                                 (string) (len=5) "email"
                                }


                                extra elements in list B:
                                ([]interface {}) (len=1) {
                                 (string) (len=6) "eautho"
                                }


                                listA:
                                ([]string) (len=3) {
                                 (string) (len=4) "name",
                                 (string) (len=2) "id",
                                 (string) (len=5) "email"
                                }


                                listB:
                                ([]string) (len=3) {
                                 (string) (len=4) "name",
                                 (string) (len=6) "eautho",
                                 (string) (len=2) "id"
                                }
                Test:           TestColumnReuse
FAIL
```

The reason:
```
func TestBytesToString(t *testing.T) {
	b := []byte{}
	b = append(b, []byte("abc")...)

	s := map[string]bool{
		BytesToString(b): true,
	}
	require.Contains(t, s, "abc")

	b = b[:0]
	b = append(b, []byte("xy")...)
	require.Equal(t, "xy", BytesToString(b))

	require.Contains(t, s, "xyc")
}

func TestSafeBytesToString(t *testing.T) {
	b := []byte{}
	b = append(b, []byte("abc")...)

	s := map[string]bool{
		string(b): true,
	}
	require.Contains(t, s, "abc")

	b = b[:0]
	b = append(b, []byte("xy")...)
	require.Equal(t, "xy", string(b))

	require.Contains(t, s, "abc")
}
```

Should we avoid reuse of ColumnAlloc.name

--- a/db_test.go
+++ b/db_test.go
@@ -16,6 +16,7 @@ import (
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/require"
 
 	"github.com/go-pg/pg/v10"
 	"github.com/go-pg/pg/v10/orm"
@@ -2510,3 +2511,55 @@ var _ = Describe("many2many multi-tenant bug", func() {
 		Expect(recipe.Ingredients[0].Id).To(Equal(1))
 	})
 })
+
+func TestColumnReuse(t *testing.T) {
+	db := testDB()
+
+	err := createSchema(db)
+	panicIf(err)
+
+	user1 := &User{
+		Name:   "admin",
+		Emails: []string{"admin1@admin", "admin2@admin"},
+	}
+	err = CreateUser(db, user1)
+	panicIf(err)
+
+	err = CreateUser(db, &User{
+		Name:   "root",
+		Emails: []string{"root1@root", "root2@root"},
+	})
+	panicIf(err)
+
+	story1 := &Story{
+		Title:    "Cool story",
+		AuthorId: user1.Id,
+	}
+	err = CreateStory(db, story1)
+	panicIf(err)
+
+	users := []map[string]interface{}{}
+	stories := []map[string]interface{}{}
+
+	ctx := context.Background()
+	_, err = db.QueryContext(ctx, &users, `SELECT * FROM users`)
+	panicIf(err)
+
+	_, err = db.QueryContext(ctx, &stories, `SELECT * FROM stories`)
+	panicIf(err)
+
+	for _, user := range users {
+		ks := []string{}
+		for k, _ := range user {
+			ks = append(ks, k)
+		}
+		require.ElementsMatch(t, []string{"name", "id", "emails"}, ks)
+	}
+	for _, story := range stories {
+		ks := []string{}
+		for k, _ := range story {
+			ks = append(ks, k)
+		}
+		require.ElementsMatch(t, []string{"id", "title", "author_id"}, ks)
+	}
+}

--- a/internal/pool/reader.go
+++ b/internal/pool/reader.go
@@ -39,7 +39,7 @@ func NewColumnAlloc() *ColumnAlloc {
 
 func (c *ColumnAlloc) Reset() {
 	c.columns = c.columns[:0]
-	c.name = c.name[:0]
+	c.name = nil
 }
 
 func (c *ColumnAlloc) New(index int16, name []byte) *ColumnInfo {

